background.js

```diff
@@ -1,128 +1,374 @@
+
+/**
+ * Populates config with option values from 3 places
+ * 1. Cloud Storage, 2. Local Storage, and 3. Default
+ * If any values are in local storage they will overwrite cloud storage values.
+ * Values should no longer be saved to local storage.
+ *
+ * @author Deshaun (From Dev Color)
+ * @param items option values retrieved from cloud storage
+ * @return None
+ */
+function fillRcxValsFromLocalStoreCloudAndDefaultValues(items) {
+	/*
+		For Each Option Value below,
+		Condition 1: Cloud Storage Value
+		Condition 2: Local Storage Value
+		Condition 3: Default Value
+	 */
+
+	if (initStorage("v0.8.92", true)) {
+		/** Useful old versioning logic */
+		// 	if (initStorage("v0.8.92", true)) {
+		// 		// v0.7
+		// 		// initStorage("popupcolor", "blue");
+		// 		initStorage("highlight", true);
+		//
+		// 		// v0.8
+		// 		// No changes to options
+		//
+		// 		// V0.8.5
+		// 		initStorage("textboxhl", false);
+		//
+		//
+		// });
+
+		// v0.8.8
+		// Backward compatibility logic may drop soon.
+		if (localStorage['highlight'] == "yes")
+			localStorage['highlight'] = "true";
+		if (localStorage['highlight'] == "no")
+			localStorage['highlight'] = "false";
+		if (localStorage['textboxhl'] == "yes")
+			localStorage['textboxhl'] = "true";
+		if (localStorage['textboxhl'] == "no")
+			localStorage['textboxhl'] = "false";
+		if (localStorage['onlyreading'] == "yes")
+			localStorage['onlyreading'] = "true";
+		if (localStorage['onlyreading'] == "no")
+			localStorage['onlyreading'] = "false";
+	}
+
+	//Todo: Need to Organize these options in alphabetical order.
+	/** Set Copy Separator Option value */
+	if (items.copySeparator) {
+		rcxMain.config.copySeparator = items.copySeparator;
+	}
+
+	if (localStorage["copySeparator"]) {
+		rcxMain.config.copySeparator = localStorage["copySeparator"];
+	}
+
+	if (!rcxMain.config.copySeparator) {
+		rcxMain.config.copySeparator = "tab"
+	}
+	echo(rcxMain.config.copySeparator);
+
+	/** Set Disable Keys option value */
+	if (items.disablekeys) {
+		rcxMain.config.disablekeys = items.disablekeys;
+	}
+
+	if (localStorage["disablekeys"]) {
+		rcxMain.config.disablekeys = localStorage["disablekeys"];
+	}
+
+	if (!rcxMain.config.disablekeys)
+		rcxMain.config.disablekeys = "false"
+
+	/** Set highlight option value */
+	if (items.highlight) {
+		rcxMain.config.highlight = items.highlight;
+	}
+
+	if (localStorage["highlight"]) {
+		rcxMain.config.highlight = localStorage["highlight"];
+	}
+
+	if (!rcxMain.config.highlight) {
+		rcxMain.config.highlight = true
+	}
+
+	/** Set kanji components option value */
+	if (items.kanjicomponents) {
+		rcxMain.config.kanjicomponents = items.kanjicomponents;
+	}
+
+	if (localStorage["kanjicomponents"]) {
+		rcxMain.config.kanjicomponents = localStorage["kanjicomponents"];
+	}
+
+	if (!rcxMain.config.kanjicomponents) {
+		rcxMain.config.kanjicomponents = "true"
+	}
+
+	/** Set kanjiinfo option values */
+	rcxMain.config.kanjiinfo = new Array(kanjiInfoLabelList.length / 2);
```

Comment:
I haven't looked close yet but I hope we can somehow reuse the logic we have in options.js for this.